nav2_bringup/bringup/launch/slam_launch.py

```diff
@@ -70,10 +73,26 @@ def generate_launch_description():
         description='Automatically startup the nav2 stack')
 
     # Nodes launching commands
+
+    # If the provided param file doesn't have slam_toolbox params, we must remove the 'params_file'
+    # LaunchConfiguration, or it will be passed automatically to slam_toolbox and will not load 
+    # the default file
+    has_slam_toolbox_params = HasNodeParams(source_file=params_file,
+                                            node_name='slam_toolbox')
+    # slam_toolbox with provided param_file
+    push_launch_config = PushLaunchConfigurations(
+            condition=UnlessCondition(has_slam_toolbox_params))
+
+    remove_params_file = UnsetLaunchConfiguration(
+            "params_file", condition=UnlessCondition(has_slam_toolbox_params))
+
     start_slam_toolbox_cmd = IncludeLaunchDescription(
```

Comment:
I don't see where slam toolbox would get the parameters, the launch arguments of this don't have the param file given.

Reply:
All LaunchConfigurations are passed automatically to included LaunchDescriptions.

It is discussed here: https://github.com/ros2/launch/issues/313

A comment by wjwwood
>  We would like to get back to this with launch. Currently an included launch description has access to all configurations of the launch file that included it.

Reply:
But I thought we talked about adding a params file field that will be given to the nodes? https://github.com/SteveMacenski/slam_toolbox/blob/foxy-devel/launch/online_sync_launch.py#L12 so if you don't give a `launch_arguments` of `params_file` then it would default to the given online sync slam?

Reply:
This solution might very well work, but it makes launch even more confusing so looking for ways to keep the patterns at least consistent with other launch files rather than introducing new patterns. If humanly possible.

Reply:
Right now, `online_sync_launch.py` will use the `params_file` of the `slam_launch.py` automatically, unless you unset it, due to the propagation of the LaunchConfiguration by default that I linked above.

If we changed `online_sync_launch.py` LaunchConfiguration name to `slam_toolbox_params_file`, then it would not be automatically passed. To pass it, it would need to be set explicitely via `launch_arguments={'use_sim_time': use_sim_time, 'slam_toolbox_params_file', params_file}.items())`

I agree, it is quite confusing. How would you like to proceed?
I can change the LaunchConfiguration name in `slam_launch.py`, but it might be a hacky solution that hides the problem.

I can explicitely pass it here, but we also need to unset it, so it becomes something like this monster:
```
    has_slam_toolbox_params = HasNodeParams(source_file=params_file,
                                            node_name='slam_toolbox')

    start_slam_toolbox_cmd = IncludeLaunchDescription(
        PythonLaunchDescriptionSource(slam_launch_file),
        condition=IfCondition(has_slam_toolbox_params),
        launch_arguments={'use_sim_time': use_sim_time, 'params_file', params_file}.items())


   # All this below is required to NOT pass the params_file automatically to the included launch description
    push_launch_config = PushLaunchConfigurations(
            condition=UnlessCondition(has_slam_toolbox_params))

    remove_params_file = UnsetLaunchConfiguration(
            "params_file", condition=UnlessCondition(has_slam_toolbox_params))

    start_slam_toolbox_cmd = IncludeLaunchDescription(
        PythonLaunchDescriptionSource(slam_launch_file),
        condition=UnlessCondition(has_slam_toolbox_params),
        launch_arguments={'use_sim_time': use_sim_time}.items())

    pop_launch_config = PopLaunchConfigurations(
            condition=UnlessCondition(has_slam_toolbox_params))







```